bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -135,8 +135,12 @@ void DbCheckpointManager::init() {
 
   monitorThread_ = std::thread([this]() {
     while (maxNumOfCheckpoints_ && !stopped_) {
+      std::unique_lock<std::mutex> lk(lockShutdown_);
       checkAndRemove();
-      std::this_thread::sleep_for(std::chrono::seconds{ReplicaConfig::instance().dbCheckpointMonitorIntervalSeconds});
+      shutdownCond_.wait_for(lk,
+                             std::chrono::seconds{ReplicaConfig::instance().dbCheckpointMonitorIntervalSeconds},
+                             [this] { return stopped_ == true; });
+      lk.unlock();
```

Comment:
redundent.
unique_lock automatically unlock the mutex it holds on distructor.
You should remove this one.
see: https://en.cppreference.com/w/cpp/thread/unique_lock/~unique_lock